packages/flutter/lib/src/cupertino/colors.dart

```diff
@@ -1013,36 +1013,25 @@ class CupertinoDynamicColor extends Color with Diagnosticable {
   /// brightness, normal contrast, [CupertinoUserInterfaceLevelData.base]
   /// elevation level).
   CupertinoDynamicColor resolveFrom(BuildContext context) {
-    Brightness brightness = Brightness.light;
-    if (_isPlatformBrightnessDependent) {
-      brightness = CupertinoTheme.maybeBrightnessOf(context) ?? Brightness.light;
-    }
-    bool isHighContrastEnabled = false;
-    if (_isHighContrastDependent) {
-      isHighContrastEnabled = MediaQuery.maybeHighContrastOf(context) ?? false;
-    }
-
-    final CupertinoUserInterfaceLevelData level = _isInterfaceElevationDependent
-      ? CupertinoUserInterfaceLevel.maybeOf(context) ?? CupertinoUserInterfaceLevelData.base
-      : CupertinoUserInterfaceLevelData.base;
-
-    final Color resolved;
-    switch (brightness) {
-      case Brightness.light:
-        switch (level) {
-          case CupertinoUserInterfaceLevelData.base:
-            resolved = isHighContrastEnabled ? highContrastColor : color;
-          case CupertinoUserInterfaceLevelData.elevated:
-            resolved = isHighContrastEnabled ? highContrastElevatedColor : elevatedColor;
-        }
-      case Brightness.dark:
-        switch (level) {
-          case CupertinoUserInterfaceLevelData.base:
-            resolved = isHighContrastEnabled ? darkHighContrastColor : darkColor;
-          case CupertinoUserInterfaceLevelData.elevated:
-            resolved = isHighContrastEnabled ? darkHighContrastElevatedColor : darkElevatedColor;
-        }
-    }
+    final bool darkTheme = _isPlatformBrightnessDependent
+      && CupertinoTheme.maybeBrightnessOf(context) == Brightness.dark;
+
+    final bool elevated = _isInterfaceElevationDependent
+      && CupertinoUserInterfaceLevel.maybeOf(context) == CupertinoUserInterfaceLevelData.elevated;
+
+    final bool highContrast = _isHighContrastDependent
+      && (MediaQuery.maybeHighContrastOf(context) ?? false);
+
+    final Color resolved = switch ((darkTheme, elevated, highContrast)) {
+      (false, false, false) => color,
+      (false, false, true)  => highContrastColor,
+      (false, true, false)  => elevatedColor,
+      (false, true, true)   => highContrastElevatedColor,
+      (true, false, false)  => darkColor,
+      (true, false, true)   => darkHighContrastColor,
+      (true, true, false)   => darkElevatedColor,
+      (true, true, true)    => darkHighContrastElevatedColor,
+    };
```

Comment:
I find this sea of false/true makes it harder to understand what is actually going on. Is there a better way to express this that preserve a little more context of what is false/true?

Reply:
I'm not a huge fan of the long `CupertinoUserInterfaceLevelData` name, but you have a good point that 3 `bool`s in a row is difficult to follow. Hopefully it looks better now!

Reply:
Yeah, this looks better. I wish Dart would support something similar to Swift where these could just be:

```
(.light, .base, false) => color,
```